src/web/js/trove/world-lib.js

```diff
@@ -749,7 +758,18 @@
               if (!stop) {
                 k2();
               } else {
-                if (extras.closeWhenStop) {
+                if (lpf) {
+                  var handler = lpf();
+                  var handler1 = handler(thisWorldIndex);
+                  handler1.onRegister(top);
+                  handler1._listener(w, oldW, function(v) { k2(); });
+                }
+
+                if (lpf) {
```

Comment:
This appears to be a fork in the control flow.  I think this should do `activationrecord.pause()`, and then call `handler1._listener` as above, and that should be the only control flow in this case.  The `closeWhenStop` and `else` cases should be unchanged, and they should all be in tail position.